packages/cardpay-subgraph/schema.graphql

```diff
@@ -456,3 +458,48 @@ type TokenPair @entity {
   token1: Token!
   swaps: [TokenSwap]!             @derivedFrom(field: "tokenPair")
 }
+
+type RewardProgramRegistrationPayment @entity {
+  id: ID!
+  rewardProgramID: String!
+  admin: Account!
+  transaction: Transaction!
+  createdAt: BigInt!
+  prepaidCardPayment: PrepaidCardPayment!
+}
+
+type RewardeeRegistrationPayment @entity {
+  id: ID!
+  rewardProgramID: String!
+  rewardee: Account!
+  transaction: Transaction!
+  createdAt: BigInt!
+  prepaidCardPayment: PrepaidCardPayment!
+}
+
+type RewardSafe @entity {
+  id: ID!
+  rewardProgramID: String!
+  rewardee: Account!
+  safe: Safe!
+}
+
+type RewardeeClaim @entity {
+  id: ID!
+  rewardProgramID: String!
+  rewardee: Account!
+  rewardSafe: RewardSafe!
+  amount: BigInt!
+  transaction: Transaction!
+  timestamp: BigInt!
+}
+
```

Comment:
I feel like you are missing an entity: `RewardProgram`. This entity would have derived field edges to all the reward safes for the program, as well as a derived field edge to all the RewardTokenAdd events.

Reply:
ok added